### PR TITLE
[JEP-230] [JENKINS-55582] Convert `instance-identity` to a detached plugin

### PR DESCRIPTION
**Pull request by jglick**
See [JEP-230](https://jenkins.io/jep/230), especially as of https://github.com/jenkinsci/jep/pull/389. Continues work done in #5304 and #6570.

Sanity testing: start Jenkins from scratch; accept default plugins; check that `/instance-identity/` endpoint works; create an inbound TCP agent (need to first enable TCP port) and connect; run a Pipeline build on it. Disable `instance-identity` despite stern warnings and restart—the agent will not work, and there are also various errors from `sshd` and `github` plugins. Enable it again and restart—back to normal. Enable SSHD and use the CLI in `-ssh` mode and via `ssh`.

### Proposed changelog entries

* The `instance-identity` module has been converted to a detached plugin.

### Proposed upgrade guidelines

Normally the new plugin will be automatically installed. If you use an “as-code” installation mechanism like a text file with a list of plugins, you may need to add `instance-identity` (version 3.1 or later) in order to restore this functionality. In particular, inbound agents using TCP (but not WebSocket) transport require this plugin to be installed. There may be other cases, which can be identified by use of APIs in the `jenkins.model.identity` package rather than directly accessing `org.jenkinsci.main.modules.instance_identity`.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6585"></a>

### CODE REVIEW

File: war/pom.xml
Comment (jglick) [inline war/pom.xml:422]: https://github.com/jenkinsci/sshd-plugin/pull/75

File: bom/pom.xml
Comment (jglick) [inline bom/pom.xml:261]: Once this is released, `bom-weekly` can get it.

File: war/pom.xml
Comment (jglick) [inline war/pom.xml:422]: https://github.com/jenkinsci/sshd-plugin/pull/75

File: war/pom.xml
Comment (jglick) [inline war/pom.xml:398]: dep of current `instance-identity`

File: bom/pom.xml
Comment (jglick) [reply bom/pom.xml:261]: Though if I understand correctly from https://github.com/jenkinsci/bom/pull/671 https://github.com/jenkinsci/bom/pull/570#issuecomment-890346549 https://github.com/jenkinsci/bom/issues/681, it cannot be added until this PR is in the _oldest_ supported line, so that will be a while. In the meantime the handful of plugins using this dep can specify 3.1.

**Comment by daniel-beck** [comment]
https://github.com/jenkinsci/jenkins/blob/f9912f5db6d5c4fd3d1895fd26b84e5c94b45f1f/core/src/main/java/jenkins/security/ResourceDomainConfiguration.java#L148-L190 will need an alternative approach for detecting whether "there" is "here".

**Comment by jglick** [comment]
> will need an alternative approach for detecting whether "there" is "here"

Not going to bother with that. If you avoid installing `instance-identity` (which is very unlikely in GUI mode) then you would just not get this bit of form validation.

File: core/src/main/resources/jenkins/split-plugins.txt
Comment (jglick) [inline core/src/main/resources/jenkins/split-plugins.txt:42]: Note: split point may need to be updated just before merge.

**Review by timja — APPROVED** [review]

**Comment by jglick** [comment]
I am still sifting through test failures in https://github.com/jenkinsci/bom/pull/1160. So far things look like they are most likely PCT issues—it does not know to convert a `provided` module dep to a detached plugin dep.

**Comment by jglick** [comment]
Finally got https://github.com/jenkinsci/bom/pull/1160 to pass by making a technical change in how PCT was run—no apparent functional regressions.

**Review by res0nance — APPROVED** [review]

File: core/src/main/resources/jenkins/split-plugins.txt
Comment (jglick) [reply core/src/main/resources/jenkins/split-plugins.txt]: (have been trying to keep this up to date when merging in `master`, but whoever finally merges this PR should double-check)

**Review by oleg-nenashev — APPROVED** [review]
+1 for pushing it forward

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/resources/jenkins/split-plugins.txt
Comment (timja) [inline core/src/main/resources/jenkins/split-plugins.txt:42]: next release is a security one so it won't be happening on this
```suggestion
instance-identity 2.356 3.1
```

**Comment by timja** [comment]
Caused ATH failure: https://github.com/jenkinsci/acceptance-test-harness/issues/796

**Comment by basil** [comment]
Likely cause of [JENKINS-70206](https://issues.jenkins.io/browse/JENKINS-70206).

**Pull request merged**
